### PR TITLE
Add quick add permissions to the cataloguer role

### PR DESCRIPTION
**Pull request by kehh**
- This overrides cataloguer in `base.xml`.
 - Adds permissions to use the quick add actions for sets, entnties, places and occurrences
 - Still cannot quick add objects, loans

### CODE REVIEW

File: profile/rwahs.xml [hunk 1]
```diff
@@ -7488,6 +7488,69 @@
...
+        <action>can_set_preferences</action>
+        <action>can_create_adv_search_forms</action>
+        <action>can_edit_adv_search_forms</action>
+        <action>can_delete_adv_search_forms</action>
+        <action>can_use_adv_search_forms</action>
+        <action>can_create_ca_bundle_displays</action>
+        <action>can_edit_ca_bundle_displays</action>
+        <action>can_delete_ca_bundle_displays</action>
+        <action>can_use_ca_bundle_displays</action>
+        <action>can_create_user_groups</action>
```
Comment (leftclickben) [inline profile/rwahs.xml]: I don't think this one is necessary for a cataloguer.

File: profile/rwahs.xml [hunk 2]
```diff
@@ -7488,6 +7488,69 @@
...
+        <action>can_search_ca_collections</action>
+        <action>can_browse_ca_collections</action>
+        <action>can_create_ca_storage_locations</action>
+        <action>can_edit_ca_storage_locations</action>
+        <action>can_delete_ca_storage_locations</action>
+        <action>can_search_ca_storage_locations</action>
+        <action>can_browse_ca_storage_locations</action>
+        <action>can_create_ca_lists</action>
+        <action>can_edit_ca_lists</action>
+        <action>can_delete_ca_lists</action>
```
Comment (leftclickben) [inline profile/rwahs.xml]: I can't help but think the `can_` prefix is a bit redundant...

File: profile/rwahs.xml [hunk 3]
```diff
@@ -7488,6 +7488,69 @@
       <actions>
       </actions>
     </role>
+    <role code="cataloguer">
+      <name>Cataloguers</name>
+      <description>Confers actions needed by those editing catalogue entries</description>
+      <actions>
```
Comment (leftclickben) [inline profile/rwahs.xml]: Should there be tests that check at least the number, if not the list, of `actions` per `role`?

File: profile/rwahs.xml (hunk 1 above)
Comment (kehh) [inline profile/rwahs.xml]: Those are different from the ACL groups, and I don't know what they do (I think it's ad hoc teams of users in a similar way to sets being ad hoc groups of objects), but I think that's fair enough - will remove.

File: profile/rwahs.xml (hunk 2 above)
Comment (kehh) [inline profile/rwahs.xml]: ya think?

File: profile/rwahs.xml (hunk 3 above)
Comment (kehh) [inline profile/rwahs.xml]: ok there you go

Comment (kehh) [inline profile/rwahs.xml]: Went for the count ;)

File: tests/Profile/ProfileACLTest.php [hunk 4]
```diff
@@ -56,4 +57,9 @@ public function testExampleUsersExist()
             $this->assertEquals(1, $this->xpath->query("/profile/logins/login[@user_name='$role_code']")->length, "An example user for the role `$role_code` should exist");
         }
     }
+
+    public function testCataloguerPermissions()
+    {
+        $this->assertEquals(56, $this->xpath->query('/profile/roles/role[@code="cataloguer"]/actions/action')->length, 'The number of actions in the cataloguer role must be equal.');
```
Comment (leftclickben) [inline tests/Profile/ProfileACLTest.php]: `'The cataloguer role should have the correct number of action permissions'`..?  Eh

**Pull request merged**